### ISSUE TITLE
I can view dynamic data in the case number column on my dashboard table

### ISSUE THREAD

**Issue by csprayregen** [pieforproviders/pieforproviders#480]
🧮 Blocked behind #481 

## ✨ What new feature do you want to see?
The case number column on my dashboard displays dynamic data.

## 🧶 Acceptance Criteria
- [x]  I see the case numbers of the children associated with my account
- [x]  The above criteria work in both English and Spanish
- [ ]  Verify accessibility (see details below)

## 🖍️ Implementation Details
Currently blocked by #428 

> **NOTE TO ENGINEERS**
> 
> Please ensure that you are running Google Lighthouse and/or WebAIM (Wave) on any UI elements in your pull request to check for accessibility issues.
> 
> If it is out of scope to address those accessibility issues in your pull request (because the issue is in a parent component and needs refactoring to work, for example), **you are responsible for noting those issues in your pull request and writing bug tickets to address the issues - make sure you link the new tickets in your pull request**

**Endpoint**: `GET /api/v1/case_list_for_dashboard`
**Authentication** required: yes
**Authorization**: scoped to the user's children
**Example response**:
```
[
    {
        "full_name": "Maria Baca",
        "approvals": [
            {
                "case_number": "5184465331"
            }
        ]
    },
    {
        "full_name": "K'Shawn Henderson",
        "approvals": [
            {
                "case_number": "5198908053"
            }
        ]
    }
]
```

## 🥧 Additional context
Figma link: https://www.figma.com/file/uCveVQQnYcw3MsenD1PNcG/P4P-2.0?node-id=2734%3A261

**Comment by cjhaddad** [comment]
@csprayregen this will be QA-able when #557 is finished
